Tests/IntegrationTests.Shared/RealmResults/SimpleLINQtests.cs

```diff
@@ -361,6 +373,91 @@ public void FirstWorks()
 
 
         [Test]
+        public void FirstOrDefaultReturnsDefault()
+        {
+            var expectedDef = _realm.All<Person>().FirstOrDefault(p => p.FirstName == "Zaphod");
+            Assert.That(expectedDef, Is.Null);
+
+            expectedDef = _realm.All<Person>().Where(p => p.FirstName == "Just Some Guy").FirstOrDefault();
+            Assert.That(expectedDef, Is.Null);
+        }
+
+
+        [Test]
+        public void LastFailsToFind()
+        {
+            Assert.Throws<InvalidOperationException>(() => _realm.All<Person>().Last(p => p.Latitude > 100));
+            Assert.Throws<InvalidOperationException>(() => _realm.All<Person>().Where(p => p.Latitude > 100).Last());
+            Assert.Throws<InvalidOperationException>(() => _realm.All<Person>().Last(p => p.Score > 50000));
+            Assert.Throws<InvalidOperationException>(() => _realm.All<Person>().Last(p => p.LastName == "Samantha"));
+        }
+
+        [Test]
+        public void LastWorks()
+        {
+            var s0 = _realm.All<Person>().Last(p => p.Longitude < -70.0 && p.Longitude > -90.0);
+            Assert.That(s0.Email, Is.EqualTo("john@doe.com"));  // Last same as First when one match
+
+            var s1 = _realm.All<Person>().Where(p => p.Score == 100.0f).Last();
+            Assert.That(s1.Email, Is.EqualTo("john@doe.com"));
+
+            var s2 = _realm.All<Person>().Last(p => p.FirstName == "John");
+            Assert.That(s2.FirstName, Is.EqualTo("John"));  // order not guaranteed in two items but know they match this
+        }
+
+
+        [Test]
+        public void LastOrDefaultReturnsDefault()
+        {
+            var expectedDef = _realm.All<Person>().LastOrDefault(p => p.FirstName == "Zaphod");
+            Assert.That(expectedDef, Is.Null);
+
+            expectedDef = _realm.All<Person>().Where(p => p.FirstName == "Just Some Guy").LastOrDefault();
+            Assert.That(expectedDef, Is.Null);
+        }
+
+
+        [Test]
+        public void ElementAtOutOfRange()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => _realm.All<Person>().Where(p => p.Latitude > 140).ElementAt(0));
```

Comment:
We could add two more tests without the `.Where` - `.ElementAt(-1)` and `.ElementAt(100000)`